numpy/random/generator.pyx

```diff
@@ -3475,21 +3492,35 @@ cdef class Generator:
 
         # GH10839, ensure double to make tol meaningful
         cov = cov.astype(np.double)
-        (u, s, v) = svd(cov)
+        if method == 'svd':
+            from numpy.dual import svd
+            (u, s, vh) = svd(cov)
+        elif method == 'eigh':
+            from numpy.dual import eigh
+            (s, u)  = eigh(cov)
+        else:
+            from numpy.dual import cholesky
+            l = cholesky(cov)
 
-        if check_valid != 'ignore':
+        if check_valid != 'ignore' and method != 'cholesky':
             if check_valid != 'warn' and check_valid != 'raise':
-                raise ValueError("check_valid must equal 'warn', 'raise', or 'ignore'")
-
-            psd = np.allclose(np.dot(v.T * s, v), cov, rtol=tol, atol=tol)
+                raise ValueError(
+                    "check_valid must equal 'warn', 'raise', or 'ignore'")
+            if method == 'svd':
+                psd = np.allclose(np.dot(u * s, u.T), cov, rtol=tol, atol=tol)
```

Comment:
I think you should be using `np.conj(v.T)`, in case one decides they want complex Gaussians.

Reply:
This function doesn't provide for a complete parameterization of complex gaussians, which need 3 cov-live arrays. Should complex be supported at all?

Reply:
I think the same code path would work except one would need to generate vectors with circularly normal distributed entries instead of standard normal.

Reply:
> Should complex be supported at all?

I'm for supporting them, if not at least one should error on invalid covariance matrices.

Reply:
We wouldn't have to worry about this if we left the original formulation in terms of vh, right?

Reply:
We would still need to generate circularly normal instead of normal. Which is easy, it's just `z1 + z2 * i`, where `z1` and `z2` are standard normal. That's the only change needed (in case the covariance is a complex `dtype`).

Reply:
There are issues related to asking for a complex normal generator.  IMO it is better to leave complex normals to a different issue/PR. See #9098.

Reply:
Agreed - let's limit discussion to things which worked before but not after this change, which might be an empty set.

Reply:
@hameerabbasi: if you feel that way, please make a new PR with the change

Reply:
See #14733.